notebooks/advanced/Time_Series_Forecasting_with_Unstructured_Data_and_Amazon_SageMaker_Neural_Topic_Model/1_preprocess.ipynb

```diff
@@ -0,0 +1,273 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Introduction"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Download the raw data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook is the first of three notebooks on combining natural language processing with time series forecasting using Amazon SageMaker and Amazon Forecast. As a first step, we download the raw dataset from UCI: the dataset consists of news articles and their headlines and titles and their source on 4 major topics. Associated sentiment scores and article ratings on Facebook, GooglePlus and LinkedIn are provided.\n",
+    "\n",
+    "The dataset can be viewed in 2 ways:\n",
+    "\n",
+    "1) Regression: given an article, predict its popularity\n",
+    "\n",
+    "2) Given a topic, forecast the popularity of the topic on various social media channels from historical data out into the future.\n",
+    "\n",
+    "Since we want to leverage Amazon Forecast, we treat it as the latter problem. A major thrust of this workshop is to demonstrate how unstructured text data can be included in Forecasting problems. That will be the topic of Notebook 2 (2_NTM.ipynb) and 3 (3_Forecast.ipynb).\n",
+    "\n",
+    "But first, we need to download the preprocess the dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import pandas as pd\n",
+    "import requests"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if os.path.exists('data/'):\n",
+    "    pass\n",
+    "else:\n",
+    "    os.mkdir('data')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "url = 'https://archive.ics.uci.edu/ml/machine-learning-databases/00432/Data/News_Final.csv'\n",
+    "r = requests.get(url, allow_redirects=True, verify=False)\n",
+    "open('data/News_Final.csv', 'wb').write(r.content)"
```

Comment:
The file is not closed after writing. Please close the file or use "with" statement to make sure file is closed.